src/main/groovy/org/kordamp/gradle/plugin/jandex/tasks/JandexTask.groovy

```diff
@@ -59,96 +50,143 @@ import javax.inject.Inject
+        processResourcesDir = objects.directoryProperty()
         destination = objects.fileProperty()
 
-        destination.convention(indexName.provider.map(new Transformer<RegularFile, String>() {
+        destination.convention(indexName.flatMap(new Transformer<Provider<RegularFile>, String>() {
             @Override
-            RegularFile transform(String s) {
-                if (resolvedIncludeInJar.get()) {
-                    File destinationDir = processResourcesTask.get().destinationDir
-                    return layout.get().projectDirectory.file("${destinationDir}/META-INF/jandex.idx".toString())
-                }
-                layout.get().buildDirectory.file('jandex/' + s).get()
+            Provider<RegularFile> transform(String s) {
+                return includeInJar.flatMap(new Transformer<Provider<RegularFile>, Boolean>() {
+                    @Override
+                    Provider<RegularFile> transform(Boolean includeInJar) {
+                        if (includeInJar) {
+                            return processResourcesDir.file("META-INF/jandex.idx")
+                        }
+                        return layout.flatMap(new Transformer<Provider<RegularFile>, ProjectLayout>() {
+                            @Override
+                            Provider<RegularFile> transform(ProjectLayout l) {
+                                return l.getBuildDirectory().file('jandex/' + s)
+                            }
+                        })
+                    }
+                })
             }
         }))
     }
 
     @Option(option = 'jandex-process-default-file-set', description = "Include the 'main' source set. Defaults to true")
-    void setProcessDefaultFileSet(boolean value) { processDefaultFileSet.property.set(value) }
+    void setProcessDefaultFileSet(boolean value) {
+        processDefaultFileSet.set(value)
+    }
 
     @Option(option = 'jandex-include-in-jar', description = "Include the generated index in the default JAR. Defaults to true")
-    void setIncludeInJar(boolean value) { includeInJar.property.set(value) }
+    void setIncludeInJar(boolean value) {
+        includeInJar.set(value)
+    }
 
     @Option(option = 'jandex-index-name', description = "The name of the index file. Defaults to jandex.idx")
-    void setIndexName(String value) { indexName.property.set(value) }
+    void setIndexName(String value) {
+        indexName.set(value)
+    }
 
     @Option(option = 'jandex-index-version', description = "The version of the index file. Defaults to the latest version supported by the invoked Jandex version")
-    void setIndexVersion(Integer value) { indexVersion.property.set(value) }
+    void setIndexVersion(Integer value) {
+        indexVersion.set(value)
+    }
 
     @Internal
-    Property<Boolean> getProcessDefaultFileSet() { processDefaultFileSet.property }
+    Property<Boolean> getProcessDefaultFileSet() { processDefaultFileSet }
```

Comment:
Given that explicit `Property<>` types are now used, these matching getters can be removed